examples/powercli/sdrs-recommendation-webexteams/handler/script.ps1

```diff
@@ -0,0 +1,37 @@
+# Process function Secrets passed in
+$VC_CONFIG_FILE = "/var/openfaas/secrets/vc-sdrs-config"
+$VC_CONFIG = (Get-Content -Raw -Path $VC_CONFIG_FILE | ConvertFrom-Json)
+if($env:function_debug -eq "true") {
+    Write-host "DEBUG: `"$VC_CONFIG`""
+}
+
+# Process payload sent from vCenter Server Event
+$json = $args | ConvertFrom-Json
+if($env:function_debug -eq "true") {
+    Write-Host "DEBUG: json=`"$($json | Format-List | Out-String)`""
+}
+
+$vcenter = ($json.source -replace "/sdk","")
+$datacenter = $json.data.datacenter.name
+$DatastoreCluster = $json.data.objectname
+
+$Body = "A new storage DRS recommendation has been generated `r`n"
+
+$Body = $Body + @"
+    vCenter Server: $vcenter
+    Datacenter: $datacenter
+    Datastore cluster: $DatastoreCluster	
+"@
+
+
+$bodytemp = @{
+    roomId=$($VC_CONFIG.WEBEX_ROOM_ID)
+    markdown=$Body
+}
+$json = $bodytemp | ConvertTo-Json
+Invoke-RestMethod -Method Post `
```

Comment:
What happens if this `Invoke-RestMehod` fails? In other functions (e.g. Python, Go) we log an error which is picked up by OpenFaaS and returned to VEBA. So VEBA understands (logs) that an error occurred.

Reply:
You can place the `Invoke-RestMethod` within try/catch block to print out the specific errors return by the cmdlet. Here's an example using `Invoke-Webrequest` https://github.com/vmware/PowerCLI-Example-Scripts/blob/master/Modules/VMware.VMC.NSXT/VMware.VMC.NSXT.psm1#L92-L107

Reply:
Added the try/catch block in commit [fc2d764](https://github.com/vmware-samples/vcenter-event-broker-appliance/pull/112/commits/fc2d7645b2fbe813b64cae02b6f18f0fc9b8c173)